numpy/core/src/common/npy_cpu_features.c

```diff
@@ -275,47 +296,61 @@ npy__cpu_try_disable_env(void)
             notsupp_cur[flen] = ' '; notsupp_cur += flen + 1;
             goto next;
         }
-        // Finally we can disable it
-        npy__cpu_have[feature_id] = 0;
+        // Finally we can disable or mark for enabling
+        npy__cpu_have[feature_id] = disable ? 0:2;
     next:
         feature = strtok(NULL, delim);
     }
+    if (!disable){
+        // Disables any unmarked dispatched feature.
+        #define NPY__CPU_DISABLE_DISPATCH_CB(FEATURE, DUMMY) \
+            if(npy__cpu_have[NPY_CAT(NPY_CPU_FEATURE_, FEATURE)] != 0)\
+            {npy__cpu_have[NPY_CAT(NPY_CPU_FEATURE_, FEATURE)]--;}\
+
+        NPY_WITH_CPU_DISPATCH_CALL(NPY__CPU_DISABLE_DISPATCH_CB, DUMMY) // extra arg for msvc
+    }
 
     *nexist_cur = '\0';
     if (nexist[0] != '\0') {
         *(nexist_cur-1) = '\0'; // trim the last space
-        if (PyErr_WarnFormat(PyExc_RuntimeWarning, 1,
-                NPY__CPU_ENV_ERR_HEAD
-                "You cannot disable CPU features (%s), since "
-                "they are not part of the dispatched optimizations\n"
-                "(" NPY_WITH_CPU_DISPATCH ").",
-                nexist
+        if (PyErr_WarnFormat(PyExc_ImportWarning, 1,
```

Comment:
```suggestion
        if (PyErr_WarnFormat(PyExc_RuntimeWarning, 1,
```
I agree with it but still I'm worry a little bit about this change, and it may require a separate release note since `PyExc_ImportWarning` is suppressed by default and the user may not be a ware. @mattip, @seberg, @rgommers  any thoughts?

Reply:
Cannot hurt to mention in the release note.  I wouldn't worry about the warning class change, myself.  Worst case to me seems 3 CI setups needing a slight adjustments (end-user will not notice this unless they are into SIMD; for them, at worst, there is a new warning).

So I think: Choose whatever you think is the best API.  This seems mainly useful for CI/testing or as a work-around for strange setups so a spurious warning seems OK anyway.

Reply:
no, I was worry that's all and I agree with @seberg.

Reply:
Thanks, then I will resolve this